### PR TITLE
Rejig JS module loading slightly

### PR DESCRIPTION
**Pull request by sndrs**
## What does this change?

- creates a `commercial*`/`enhanced` commons chunk in webpack
- loads `commercial*`/`enhanced` sooner
- runs `commercial*`/`enhanced` in parallel

also

- makes webpack's `import` usable
- tidies up `boot.js`

## What is the value of this and can you measure success?

`enhanced` no longer needs to wait for `commercial` js to complete, without delaying commercial (hopefully running it sooner).

where the preload header operates this will not make a lot of difference, but where we rely on the JS to load the rest of it, it should.

## before
screen shot 2018-01-26 at 11 37 35

## after
screen shot 2018-01-26 at 11 41 25

### CODE REVIEW

**Comment by rich-nguyen** [comment]
Why not just combine `commercial` and `enhanced`?

File: webpack.config.js [hunk 1]
```diff
@@ -109,6 +109,13 @@ module.exports = {
             // add errors to webpack instead of warnings
             failOnError: true,
         }),
+
+        new webpack.optimize.CommonsChunkPlugin({
+            name: 'standard',
```
Comment (rich-nguyen) [inline webpack.config.js]: I actually tried this and was concerned I was front loading `standard` with modules it does not need. And I didn't fancy creating a separate commons chunk either, for fear of an overall delay. How big is standard after this?

File: webpack.config.prod.js [hunk 2]
```diff
@@ -16,11 +16,6 @@ module.exports = webpackMerge.smart(config, {
     },
     devtool: 'source-map',
     plugins: [
-        new webpack.optimize.AggressiveMergingPlugin({
```
Comment (rich-nguyen) [inline webpack.config.prod.js]: 👍

File: webpack.config.js (hunk 1 above)
Comment (sndrs) [reply webpack.config.js]: standard doesn't change, that just for webpack so it knows the chunks are children of that particular entry point (we only have one entry point).

you can see in the screen shots that effectively a load of common stuff is moved into that `0` chunk

File: webpack.config.prod.js (hunk 2 above)
Comment (rich-nguyen) [reply webpack.config.prod.js]: I like this idea, but when I tried it, I think it caused a number of discrete smaller chunks to be created, that probably did feel right to aggressively merge. Do you have more chunks after removing this plugin? I just hate that magic number!

File: webpack.config.js (hunk 1 above)
Comment (sndrs) [reply webpack.config.js]: from commercial...

**Review by rich-nguyen — COMMENTED** [review]
I think the idea is great, but I'm unsure if downloading the same content as two chunks in parallel is faster than downloading a single chunk, h2 or otherwise.

File: webpack.config.js (hunk 1 above)
Comment (rich-nguyen) [reply webpack.config.js]: I thought that if you specify an existing chunk as `name`, then this named chunk becomes the commons chunk, and has the additional commons content appended to it?

Comment (rich-nguyen) [reply webpack.config.js]: ah, but then you've set `async`. you should use `async: 'commons'`

Comment (sndrs) [reply webpack.config.js]: woah, what's that?

Comment (sndrs) [reply webpack.config.js]: image

here's the commons chunk

**Comment by sndrs** [comment]
> I think the idea is great, but I'm unsure if downloading the same content as two chunks in parallel is faster than downloading a single chunk, h2 or otherwise.

i don't think it should affect commercial, but should ensure enahcned can run sooner?

**Comment by sndrs** [comment]
closing for a further proposal

**Pull request closed without merging**